app/assets/src/components/views/compare/SamplesHeatmapVis.jsx

```diff
@@ -0,0 +1,155 @@
+      { key: "NT.r", label: "NT r (total reads)" },
+      { key: "NR.zscore", label: "NR Z Score" },
+      { key: "NR.rpm", label: "NR rPM" },
+      { key: "NR.r", label: "NR r (total reads)" }
+    ];
+
+    this.handleCellClick = this.handleCellClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.heatmap = new Heatmap(
+      this.heatmapContainer,
+      {
+        rowLabels: this.extractTaxonLabels(),
+        columnLabels: this.extractSampleLabels(),
+        values: this.props.data[this.props.metric]
+      },
+      {
+        scale: this.props.scale,
+        tooltipContainer: this.tooltipContainer,
+        onNodeHover: this.handleNodeHover,
+        onRemoveRow: this.props.onRemoveTaxon,
+        onColumnLabelClick: this.props.onSampleLabelClick,
+        onCellClick: this.handleCellClick
+      }
+    );
+  }
+
+  componentDidUpdate() {
+    if (this.props.scale !== this.scale) {
+      this.scale = this.props.scale;
+      this.heatmap.updateScale(this.props.scale);
+    }
+  }
+
+  extractSampleLabels() {
+    return this.props.sampleIds.map(id => this.props.sampleDetails[id].name);
+  }
+
+  extractTaxonLabels() {
+    return this.props.taxonIds.map(id => this.props.taxonDetails[id].name);
+  }
+
+  handleNodeHover(node) {
+    this.setState({ nodeHover: node });
+  }
+
+  download() {
+    this.heatmap.download();
+  }
+
+  getTooltipData() {
+    let node = this.state.nodeHover;
+    let sampleId = this.props.sampleIds[node.columnIndex];
+    let taxonId = this.props.taxonIds[node.rowIndex];
+    let sampleDetails = this.props.sampleDetails[sampleId];
+    let taxonDetails = this.props.taxonDetails[taxonId];
+
+    let nodeHasData = this.metrics.some(
+      metric => !!this.props.data[metric.key][node.rowIndex][node.columnIndex]
+    );
+
+    let values = null;
+    if (nodeHasData) {
+      values = this.metrics.map(metric => {
+        let data = this.props.data[metric.key];
+        let value = (data[node.rowIndex][node.columnIndex] || 0).toFixed(0);
+        return [
+          metric.label,
+          metric.key === this.props.metric ? <b>{value}</b> : value
+        ];
+      });
+    }
+
+    return [
+      {
+        name: "Info",
+        data: [
+          ["Sample", sampleDetails.name],
+          ["Taxon", taxonDetails.name],
+          ["Category", taxonDetails.category]
+        ]
+      },
+      {
+        name: "Values",
+        data: nodeHasData ? values : [["", "Taxon not found in Sample"]]
+      }
+    ];
+  }
+
+  handleCellClick(cell) {
+    openUrl(`/samples/${this.props.sampleIds[cell.columnIndex]}`, event);
+  }
+
+  render() {
+    return (
+      <div className="samples-heatmap-vis">
+        <div
+          className="samples-heatmap-vis__container"
+          ref={container => {
+            this.heatmapContainer = container;
+          }}
+        />
+        <div
+          id="heatmap"
+          className="samples-heatmap-vis__tooltip"
+          ref={tooltip => {
```

Comment:
Should we use `React.createRef` for consistency?

Reply:
This is actually the consistent way.... and the way i like for d3 integration.
I might change the createref i used in the `SamplesHeatmapView`

Reply:
done